backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/pageindicator/internal/BpkPageIndicatorImpl.kt

```diff
@@ -0,0 +1,123 @@
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.pager.ExperimentalPagerApi
+import com.google.accompanist.pager.HorizontalPager
+import com.google.accompanist.pager.PagerState
+import net.skyscanner.backpack.compose.pageindicator.BpkPageIndicatorStyle
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import java.lang.Integer.min
+
+@OptIn(ExperimentalPagerApi::class)
+@Composable
+internal fun BpkPageIndicatorImpl(
+  pagerState: PagerState,
+  totalIndicators: Int,
+  indicatorLabel: String,
+  modifier: Modifier = Modifier,
+  style: BpkPageIndicatorStyle,
+) {
+
+  val listState = rememberLazyListState()
+
+  val indicatorSize = BpkSpacing.Md
+  val indicatorCount = min(totalIndicators, PAGE_INDICATOR_DOT_MAX_COUNT)
+  val totalWidth = indicatorSize * (2 * indicatorCount - 1)
+  val widthInPx = LocalDensity.current.run { totalWidth.toPx() }
+
+  val currentItem by remember { derivedStateOf { pagerState.currentPage } }
+  val itemCount = pagerState.pageCount
+
+  LaunchedEffect(key1 = currentItem) {
+    val viewportSize = listState.layoutInfo.viewportSize
+    listState.animateScrollToItem(
+      currentItem,
+      (widthInPx / 2 - viewportSize.width / 2).toInt()
+    )
+  }
+
+  Box(
+    modifier = Modifier
+      .width(totalWidth + indicatorSize * 2)
+      .height(indicatorSize * 3)
+      .semantics { contentDescription = indicatorLabel },
+    contentAlignment = Alignment.Center,
+  ) {
+    HorizontalPager(
+      count = totalIndicators,
+      state = pagerState,
+      modifier = Modifier
+        .size(1.dp)
+        .alpha(0f),
+      content = {}
+    )
+
+    LazyRow(
+      modifier = modifier.width(totalWidth),
+      state = listState,
+      horizontalArrangement = Arrangement.spacedBy(indicatorSize),
+      userScrollEnabled = false
+    ) {
+
+      items(itemCount) { index ->
+        val isSelected = (index == currentItem)
+
+        Box(
+          modifier = Modifier
+            .clip(CircleShape)
+            .size(indicatorSize)
+            .background(
+              color = if (isSelected) {
```

Comment:
The color can be animated with animateColorAsState() – for insance, see BpkText color animation in BpkNudger